gensim/models/word2vec.py

```diff
@@ -1229,6 +1209,23 @@ def prepare_vocab(self, hs, negative, wv, update=False, keep_raw_vocab=False, tr
         sample = sample or self.sample
         drop_total = drop_unique = 0
 
+        # if max_vocab is specified instead of min_count
+        # pick a min_count which satisfies max_vocab as well as possible
+        if self.max_vocab is not None:
+            import operator
+
+            sorted_vocab = sorted(self.raw_vocab.items(), key=operator.itemgetter(1), reverse=True)
+            curr_count = 0
+            calc_min_count = 0
+
+            for item in sorted_vocab:
+                curr_count += item[1]
+                if curr_count < self.max_vocab:
+                    calc_min_count = item[1]
+                else:
+                    break
+            min_count = calc_min_count
```

Comment:
This clobbers any other `min_count` provided – rather than respecting both `min_count` and `max_vocab` if both are supplied. As per my comment in #465, "If both a min_count and max_vocab are specified, they should both be satisfied - which in practice would mean whichever implies the higher min_count."

Reply:
`prepare_vocab()` logging & return-value should provide the same visibility into this parameter's effects (including in a 'dry_run`) as is available for `min_count`.

Reply:
I have changed the lines to:
```
            if calc_min_count > min_count:
                min_count = calc_min_count
```

Reply:
>prepare_vocab() logging & return-value should provide the same visibility into this parameter's effects (including in a 'dry_run) as is available formin_count`.

Do you mean I should add comments describing `max_vocab` and logging comments describing the outcome of the max_vocab processing?